Update bandit.yml with version bump

reactive-firewall opened the pull request:
## Migrate to latest, maintained, version of bandit code-scanning action.

### Noteworthy changes:
 * Maintained action is a fork of the unmaintained previous version. - Versions are still pinnable ( including bug-for-bug compatible `v1.0` ) - Maintained project now utilizes @dependabot to keep sub-dependencies current.
 * Maintained action is already released on marketplace.
 * Credits updated with both original and maintainer with no change to licensing.
 * Updated to use checkout@v4 now.

---

<details>
<summary>📋 TL;DR - PR Template with checklist from code owners</summary>

### **Please note that at this time we are only accepting new starter workflows for Code Scanning. Updates to existing starter workflows are fine.**

---

## Tasks

**For _all_ workflows, the workflow:**

- [x] Should be contained in a `.yml` file with the language or platform as its filename, in lower, [_kebab-cased_](https://en.wikipedia.org/wiki/Kebab_case) format (for example, [`docker-image.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-image.yml)).  Special characters should be removed or replaced with words as appropriate (for example, "dotnet" instead of ".NET").
- [x] Should use sentence case for the names of workflows and steps (for example, "Run tests").
- [x] Should be named _only_ by the name of the language or platform (for example, "Go", not "Go CI" or "Go Build").
- [x] Should include comments in the workflow for any parts that are not obvious or could use clarification.
- [x] Should specify least privileged [permissions](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token) for `GITHUB_TOKEN` so that the workflow runs successfully.

**For _Code Scanning_ workflows, the workflow:**

- [x] Should be preserved under [the `code-scanning` directory](https://github.com/actions/starter-workflows/tree/main/code-scanning).
- [x] Should include a matching `code-scanning/properties/*.properties.json` file (for example, [`code-scanning/properties/codeql.properties.json`](https://github.com/actions/starter-workflows/blob/main/code-scanning/properties/codeql.properties.json)), with properties set as follows:
  - [x] `name`: Name of the Code Scanning integration.
  - [ ] `creator`: Name of the organization/user producing the Code Scanning integration.
  **PLEASE ADVISE** [does this need to be changed?](https://github.com/reactive-firewall/starter-workflows/blob/main/code-scanning/properties/bandit.properties.json#L3)
  - [x] `description`: Short description of the Code Scanning integration.
  - [x] `categories`: Array of languages supported by the Code Scanning integration.
  - [x] `iconName`: Name of the SVG logo representing the Code Scanning integration. This SVG logo must be present in [the `icons` directory](https://github.com/actions/starter-workflows/tree/main/icons).
- [x] Should run on `push` to `branches: [ $default-branch, $protected-branches ]` and `pull_request` to `branches: [ $default-branch ]`. We also recommend a `schedule` trigger of `cron: $cron-weekly` (for example, [`codeql.yml`](https://github.com/actions/starter-workflows/blob/c59b62dee0eae1f9f368b7011cf05c2fc42cf084/code-scanning/codeql.yml#L14-L21)).

**Some general notes:**

- [x] This workflow must _only_ use actions that are produced by GitHub, [in the `actions` organization](https://github.com/actions), **or**
- [x] This workflow must _only_ use actions that are produced by the language or ecosystem that the workflow supports.  These actions must be [published to the GitHub Marketplace](https://github.com/marketplace?type=actions).  We require that these actions be referenced using the full 40 character hash of the action's commit instead of a tag.  Additionally, workflows must include the following comment at the top of the workflow file:
    ```
    # This workflow uses actions that are not certified by GitHub.
    # They are provided by a third-party and are governed by
    # separate terms of service, privacy policy, and support
    # documentation.
    ```
- [x] Automation and CI workflows should not send data to any 3rd party service except for the purposes of installing dependencies.
- [x] Automation and CI workflows cannot be dependent on a paid service or product.

</details>

reactive-firewall committed:
Update bandit.yml with a maintained fork

Migrate to latest, maintained, version of action.
 * Maintained action is a fork of the unmaintained previous version.
    - Versions are still pinnable ( including bug-for-bug compatible `v1.0` )
    - Maintained project now utilizes @dependabot to keep sub-dependencies current.
    - Maintained action is already released on marketplace
 * Credits both original and maintainer with no change to licensing
 * Updated to use checkout@v4 already

reactive-firewall reviewed (commented):
After Reviewing the contributor guide, it is clear that at-least the versioning will need to be changed so I've included some additional comments so that after a review by project maintainers all required changes can hopefully be selected and applied as a single batch. (Obviously these review comments may be resolved by the maintainers without changes when appropriate too)

Hope this helps

reactive-firewall commented:
Tagging possible community stakeholders ( i.e. _Request for Comment_ )

Regarding [which version to pin](https://github.com/actions/starter-workflows/pull/2497#discussion_r1752891477) for `actions/checkout` (i.e. [actions/checkout@4.1.6](https://github.com/actions/checkout/commit/a5ac7e51b41094c92402da3b24376905380afc29) or [actions/checkout@4.1.7](https://github.com/actions/checkout/commit/692973e3d937129bcbf40652eb9f2f61becf3332))
* @cory-miller
* @joshmgross


Regarding Included fixes from overlapping, related, or similar work (i.e. relevant community):
 * @abirismyname [Upstream Author of included](https://github.com/abirismyname) of [shundor/python-bandit-scan](https://github.com/shundor/python-bandit-scan)
 * Kenta Nakase Author of [included work](https://github.com/shundor/python-bandit-scan/pull/5) ([already responded, declining further interest](https://github.com/parroty/python-bandit-scan/pull/1) :bow: Please respect)
 * @Grisolfi Author of [overlapping/similar work](https://github.com/shundor/python-bandit-scan/pull/2)
 * @SamMorrowDrums Author of [overlapping/similar work](https://github.com/shundor/python-bandit-scan/pull/3)

Possible Overlap with other PRs in this repo:
* PR #2465 - this PR #2497 aims to eliminate part of the issue raised, as it concerns [bandit.yml](https://github.com/actions/starter-workflows/pull/2497#discussion_r1752891477) :bow: Coordination requested.
* PR #2464 - related changes, but does not conflict (included only for completeness)

jsoref commented on code-scanning/bandit.yml:
```diff
@@ -20,29 +20,31 @@ on:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
```
There's a note somewhere that explains that GitHub owned actions don't need to / generally shouldn't be sha pinned.

It also applies to GitHub partner owned actions. Note I have no idea how to get that list. Thankfully github.com/actions and github.com/github are cases that fall into the simpler category.

reactive-firewall replied:
@jsoref, first, thanks for taking the time to review. I'll try to find said note and review. The suggestion is harmless until applied, but remains handy if needed. :shrug:

🙊 TL;DR:
I only know the PR template/guidelines only require when NOT part of the _"github/`actions` organization"_. (from "general notes" of the [`.github/pull_request_template.md`](https://github.com/actions/starter-workflows/blob/main/.github/pull_request_template.md?plain=1#L52-L53) for referance)
Currently I'm going by the principle of consistency. It is currently consistent with how it was before; by tag ie `v2` --> `v4` ) the suggestion is for the required reviewers convenience 🙉 (as I already mentioned any v4 seems reasonable to me) Part of my concern is that the guidelines do not address how the project maintainers wish to pin official actions from 
> GitHub, i.e. _"github/`actions` organization"_

It simply states that they _can_ be used. So while I completely agree that it's probably fine, but I felt I should ask beforehand :shrug:

I'll probably just need to wait for an approval or change request from the project maintainers before making a move.

jsoref replied:
The English/logic is wrong there. But, as you're seeing, getting fixes to this repository approved is not remotely fast. 

I'm not even sure, offhand, how to fix that. The problem is the `or` at the end of the first checkbox. 

The goal is to say:
1. Workflows ideally only use GitHub's own actions. These actions may be referenced by their tags.
2. If additional actions are needed, they should be limited to those of the ecosystem and an extra warning blob included. these actions should be referenced by full SHAs.

Anyway, at some point, maybe we can get that guideline text fixed.

It doesn't even mention the partner exception. (The guidelines are as mess and the main reason I haven't bothered to create a PR to this repository for my own workflow-action.)

reactive-firewall replied:
@jsoref, I agree with your assessment completely. 🙉 I also echo you on:
> maybe we can get that guideline text fixed.

Especially considering the clout this repository has in the wider community.

Anyway, Thanks again for taking the time to give feedback. :bow: _(what is the old adage? "Optimism is an occupational hazard of programming; feedback is the cure" or something like that)_

reactive-firewall committed:
Apply trivial suggestions from contributor guidelines

Two minor changes here:
* removed unnecessary whitespace
* pined by SHA latest version of python-bandit-scan (v2.1) reactive-firewall@python-bandit-scan@c8b1d56a3964de4e00e7a820dddb38661a4b7566

jsoref commented on code-scanning/bandit.yml:
```diff
@@ -20,29 +20,30 @@ on:
+      - uses: actions/checkout@v4
       - name: Bandit Scan
-        uses: shundor/python-bandit-scan@9cc5aa4a006482b8a7f91134412df6772dbda22c
+        uses: reactive-firewall/python-bandit-scan@ c8b1d56a3964de4e00e7a820dddb38661a4b7566  # v2.1 - c8b1d56a3964de4e00e7a820dddb38661a4b7566
```
```suggestion
        uses: reactive-firewall/python-bandit-scan@c8b1d56a3964de4e00e7a820dddb38661a4b7566  # v2.1 - c8b1d56a3964de4e00e7a820dddb38661a4b7566
```

reactive-firewall replied:
good catch, the space must have been in the copy-paste, 🙈

reactive-firewall committed:
Update code-scanning/bandit.yml to fix regression

- removes an un-intended space from the config to fix the syntax. :see_no_evil:

Co-authored-by: Josh Soref <2119212+jsoref@users.noreply.github.com>

reactive-firewall reviewed (commented):
LGTM after rebase.

TL;DR - Fixed issue with space when pinning by hash instead of tag. Also rebased with latest version from default branch to ensure compatibility.

reactive-firewall reviewed (commented):
💁🏻 LGTM! No conflicts from rebasing (from `actions:main` to remain current).

reactive-firewall commented:
Another rebase to keep this current with `main`.
> 💁🏻 LGTM! No conflicts from rebasing (from `actions:main` to remain current).

hoseinrostamii commented:
من اهل کشور ایرانم برای همون ثبت نمیکنه من الان چرا باید پول بدم این چه
قانونیه

در تاریخ شنبه ۱۴ دسامبر ۲۰۲۴،‏ ۹:۵۶ Mr. Walls ***@***.***>
نوشت:

> Another rebase to keep this current with main.
>
> 💁🏻 LGTM! No conflicts from rebasing (from actions:main to remain
> current).
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/actions/starter-workflows/pull/2497#issuecomment-2542901784>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BHTDWUXJFBLAC3II5XJLQXD2FPFQJAVCNFSM6AAAAABN7WISW2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKNBSHEYDCNZYGQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

hoseinrostamii commented:
با سند و مدرک بهتون نشون میدم

در تاریخ شنبه ۱۴ دسامبر ۲۰۲۴،‏ ۱۰:۲۴ Karen service Namayandegi <
***@***.***> نوشت:

> من اهل کشور ایرانم برای همون ثبت نمیکنه من الان چرا باید پول بدم این چه
> قانونیه
>
> در تاریخ شنبه ۱۴ دسامبر ۲۰۲۴،‏ ۹:۵۶ Mr. Walls ***@***.***>
> نوشت:
>
>> Another rebase to keep this current with main.
>>
>> 💁🏻 LGTM! No conflicts from rebasing (from actions:main to remain
>> current).
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/actions/starter-workflows/pull/2497#issuecomment-2542901784>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/BHTDWUXJFBLAC3II5XJLQXD2FPFQJAVCNFSM6AAAAABN7WISW2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKNBSHEYDCNZYGQ>
>> .
>> You are receiving this because you are subscribed to this thread.Message
>> ID: ***@***.***>
>>
>

hoseinrostamii commented:
> من اهل کشور ایرانم برای همون ثبت نام نمیکنم من الان چرا باید پول بدم این چه قانونی در تاریخ شنبه ۱۴ دسامبر ۲۰۲۴، ۹:۵۶ Mr. Walls ***@***.***> نوشت:
> […](#)
> Another rebase to keep this current with main. 💁🏻 LGTM! No conflicts from rebasing (from actions:main to remain current). — Reply to this email directly, view it on GitHub <[#2497 (comment)](https://github.com/actions/starter-workflows/pull/2497#issuecomment-2542901784)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/BHTDWUXJFBLAC3II5XJLQXD2FPFQJAVCNFSM6AAAAABN7WISW2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKNBSHEYDCNZYGQ> . You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

الان چی بگم به شما خدا شاهده

reactive-firewall commented:
@alexisabril is there a chance of someone looking this PR anytime soon?

JamesonDiamond1 reviewed (approved):
@JamesonDiamond1

reactive-firewall committed:
Merge branch 'main' into patch-1

reactive-firewall commented:
@actions/actions-workflow-development-reviewers is there a chance of someone looking this PR anytime soon?

Art39print commented:
> @actions/actions-workflow-development-reviewers is there a chance of someone looking this PR anytime soon?
